gaia-component.js

```diff
@@ -400,6 +434,7 @@ function addDirObserver() {
 
   function onChanged(mutations) {
     document.dispatchEvent(new Event('dirchanged'));
+    dirRegisteredComponents.forEach(setInnerDirAttributes);
   }
```

Comment:
Previously we were binding to an event inside the component and unbinding when the component was `detached`. This meant that we didn't have to keep an external registry of `dirRegisteredComponents`. We can still do this:

```js
attachedCallback() {
   ...
  if (this.dirObserver) {
    this.setInnerDirAttributes = setInnerDirAttributes.bind(null, this);
    addEventListener('dirchanged', this.setInnerDirAttributes);
    setInnerDirAttributes(this); 
  }
  ...
},

detachedCallback() {
   ...
  if (this.dirObserver) {
    removeEventListener('dirchanged', this.setInnerDirAttributes);
  }
  ...
}
```

---

I was hoping we could use the `'localized'` event instead of the `MutationObserver` (if l10n.js present) as discussed offline, but happy to add this as a followup if we're pushed for time.